### PR TITLE
Codefix

### LINKED ISSUE #165: Add codefix for S3235 (redundant parentheses)

### PR DESCRIPTION
**Pull request by tvsonar**
Fixes #165

### CODE REVIEW

**Comment by tvsonar** [comment]
@dbolkensteyn I rebased this PR to ease the review and to contain all the latest changes.

File: src/SonarQube.SonarLint.Descriptor/packages.config [hunk 1]
```diff
@@ -1,9 +1,9 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
```
Comment (dbolkensteyn) [inline src/SonarQube.SonarLint.Descriptor/packages.config]: @tvsonar looks like an unrelated change?

File: src/SonarQube.SonarLint.Descriptor/app.config [hunk 2]
```diff
@@ -29,6 +29,6 @@
     </assemblyBinding>
   </runtime>
   <startup>
```
Comment (dbolkensteyn) [inline src/SonarQube.SonarLint.Descriptor/app.config]: @tvsonar looks like an unrelated change?

File: src/SonarLint/packages.config [hunk 3]
```diff
@@ -1,9 +1,11 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
```
Comment (dbolkensteyn) [inline src/SonarLint/packages.config]: @tvsonar looks like an unrelated change?

File: src/SonarLint/SonarLint.csproj [hunk 4]
```diff
@@ -9,7 +9,7 @@
     <AppDesignerFolder>Properties</AppDesignerFolder>
     <RootNamespace>SonarLint</RootNamespace>
     <AssemblyName>SonarLint</AssemblyName>
```
Comment (dbolkensteyn) [inline src/SonarLint/SonarLint.csproj]: @tvsonar looks like an unrelated change?

File: src/Tests/SonarLint.UnitTest/Rules/AssignmentInsideSubExpressionTest.cs [hunk 5]
```diff
@@ -30,7 +30,7 @@ public class AssignmentInsideSubExpressionTest
         [TestCategory("Rule")]
         public void AssignmentInsideSubExpression()
         {
```
Comment (dbolkensteyn) [inline src/Tests/SonarLint.UnitTest/Rules/AssignmentInsideSubExpressionTest.cs]: @tvsonar let's move this renaming to another PR - PR should be kept small to be easily reviewed

File: src/SonarLint.DocGenerator/packages.config [hunk 6]
```diff
@@ -1,9 +1,9 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
-  <package id="Microsoft.CodeAnalysis.Analyzers" version="1.0.0" targetFramework="net45" userInstalled="true" />
```
Comment (dbolkensteyn) [inline src/SonarLint.DocGenerator/packages.config]: @tvsonar looks like an unrelated change

**Comment by dbolkensteyn** [comment]
I've tested the codefix which is really nice! It works exactly as expected, and definitely improves the experience!

However please rework the PR to reduce it to a manageable size:

There is nothing worse when starting to review a PR to have the page load slowly and then see tons of unrelated changes.

It's also a good idea to run `git diff` or have a quick look at the PR to verify that no unwanted change has been included, and also that nothing has been forgotten.

**Comment by dbolkensteyn** [comment]
Also there should be an issue associated to this PR which provided a codefix for rule S3235.

File: src/Tests/SonarLint.UnitTest/Rules/AssignmentInsideSubExpressionTest.cs (hunk 5 above)
Comment (tvsonar) [inline src/Tests/SonarLint.UnitTest/Rules/AssignmentInsideSubExpressionTest.cs]: @dbolkensteyn a moved the renaming to another commit in this PR.

File: src/SonarLint.DocGenerator/packages.config (hunk 6 above)
Comment (tvsonar) [inline src/SonarLint.DocGenerator/packages.config]: @dbolkensteyn All the packages.config and app.config changes are cause by removing and readding the nuget packages to the solution. I needed to add the _Microsoft.CodeAnalysis.CSharp.Workspaces_ and _Microsoft.CodeAnalysis.Workspaces.Common_ packages, and now the `userInstalled="true"` flag is not added by VS.

**Comment by dbolkensteyn** [comment]
LGTM
